spirv_msl.cpp

```diff
@@ -11959,7 +11959,7 @@ string CompilerMSL::member_attribute_qualifier(const SPIRType &type, uint32_t in
 		else
 			quals = member_location_attribute_qualifier(type, index);
 
-		if (builtin == BuiltInBaryCoordKHR || builtin == BuiltInBaryCoordNoPerspKHR)
+		if (builtin == BuiltInBaryCoordKHR)
 		{
 			if (has_member_decoration(type.self, index, DecorationFlat) ||
 			    has_member_decoration(type.self, index, DecorationCentroid) ||
```

Comment:
This looks wrong.

Reply:
Thanks for the comment.

Without my changes, when compiling GLSL shaders to SPIR-V which has `gl_BaryCoordNoPersp`, then using SPIRV-Cross to generate Metal shaders, this error will be displayed:
```
Flat, Centroid, Sample, NoPerspective decorations are not supported for BaryCoord inputs.
```
Which is not entirely true because Metal supports `Centroid` and `NoPerspective` decorations for `BaryCoord` input according to this spec:
https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf
at page 103:

> A fragment function input declared with the `[[barycentric_coord]]` attribute can only be
> declared with either the `center_perspective` (default) or `center_no_perspective`
> sampling and interpolation attributes. The barycentric coordinates and per-pixel primitive ID
> can be passed as fragment function input in structures organized as shown in these examples:
> ```
> struct FragmentInput0 {
>     uint primitive_id [[primitive_id]];
>     // [[center_perspective]] is the default, so it can be omitted.
>     float3 barycentric_coord [[barycentric_coord, center_perspective]];
> };
> ```
> ```
> struct FragmentInput1 {
>     uint primitive_id [[primitive_id]];
>     float2 linear_barycentric_coord [[barycentric_coord,
>     center_no_perspective]];
> };
> ```

So there's clearly a bug there; My changes may not be 100% perfect but they work in my use case.
You may take a look further and advice accordingly.
PS: This PR will complement this: https://github.com/KhronosGroup/glslang/pull/3198